include/zephyr/bluetooth/conn.h

```diff
@@ -1055,6 +1057,28 @@ struct bt_conn_cb {
 	struct bt_conn_cb *_next;
 };
 
+/** @brief LTK request hook.
+ *
+ * This hook is called when the host receives a
+ * HCI_LE_Long_Term_Key_Request event.
+ *
+ * @param evt Raw event data. Can be parsed as @ref bt_hci_evt_le_ltk_request.
+ * @retval false The hook defers to the default handler.
+ * @retval true The hook takes responsibility for handling this event.
+ */
+typedef bool (*bt_conn_ltk_request_hook_t)(const uint8_t *evt);
+
+/** @brief Install the LTK request hook.
+ *
+ * Only one hook can be installed.
+ *
+ * @param cb Hook to install.
+ *
+ * @retval 0 Success
+ * @retval -ENOSPC A hook was already installed
+ */
+int bt_conn_ltk_request_hook_install(bt_conn_ltk_request_hook_t cb);
```

Comment:
Would it be possible to register it per Bluetooth LE connection? It would be beneficial for our use-case.